From fef71184c50b0a47beb655efbfd916305a2e0cad Mon Sep 17 00:00:00 2001
From: Julien Thomas <julthomas@free.fr>
Date: Thu, 30 Jan 2020 13:06:41 +0100
Subject: [PATCH] cef: fix parsing of the first extension key name

CEF format does not require a space between the last header field
separator and the first key name of the extensions, eg:

CEF:0|a|b|c|d|e|f|field1=value1 field2=value2

The cefGetHdrField() function already eats the ending separator '|'
after an header field. The ++i increment was loosing the first character
of the first extension key name.
---
 src/parser.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/parser.c b/src/parser.c
index 7343d84..932b8be 100644
--- a/src/parser.c
+++ b/src/parser.c
@@ -2840,7 +2840,6 @@ PARSER_Parse(CEF)
 	CHKR(cefGetHdrField(npb, &i, (value == NULL) ? NULL : &sigID));
 	CHKR(cefGetHdrField(npb, &i, (value == NULL) ? NULL : &name));
 	CHKR(cefGetHdrField(npb, &i, (value == NULL) ? NULL : &severity));
-	++i; /* skip over terminal '|' */
 
 	/* OK, we now know we have a good header. Now, we need
 	 * to process extensions.
-- 
2.21.1

